core/storage/user/gae_models.py

```diff
@@ -2215,3 +2215,68 @@ def get_new_id(cls, unused_entity_name):
                 return new_id
 
         raise Exception('New id generator is producing too many collisions.')
+
+
+class UserAuthModel(base_models.BaseModel):
+    """Stores the authentication details for a particular user.
+
+    Instances of this class are keyed by user id.
+    """
+
+    # Authentication detail for sign in using google id (GAE).
+    gae_id = ndb.StringProperty(indexed=True, repeated=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """The model can be deleted since it only contains information
+        relevant to one user account.
+        """
+        return base_models.DELETION_POLICY.DELETE
+
+    @staticmethod
+    def get_export_policy():
+        """Currently, the model hold authentication details relavant only for
```

Comment:
...the model holds...

... relevant only ...

Reply:
Done.